### DISCUSSION TITLE
Returning HTTP Status Code from a Route

### DISCUSSION THREAD

**Discussion by ramzerof** [remix-run/react-router#8435] [Q&A]
It is possible to return a Status Code form a Route?
classic example is a PageNotFound component that returns a 404 status code

**Comment by timdorr** [comment]
That would be up to your server implementation. React Router doesn't deal with responses or status codes.

**Reply by ramzerof** [reply]
This mean that if someone land on a page that does not exists I can't reply with a 404 at the app level,  how do people normally deal with this kind issues without some sort of replication of logic? 
Can somone provide any implementation example?

**Reply by atefBB** [reply]
If this package is based on `history` API, it doesn't deal with `HTTP` status code at all. see [this](https://developer.mozilla.org/en-US/docs/Web/API/History_API).
